src/OrchardCore/OrchardCore.Abstractions/Shell/Scope/ShellScope.cs

```diff
@@ -396,7 +408,7 @@ internal async Task ActivateShellInternalAsync()
         /// <summary>
         /// Registers a delegate to be invoked before the current shell scope will be disposed.
         /// </summary>
-        public static void RegisterBeforeDispose(Func<ShellScope, Task> callback) => Current?.BeforeDispose(callback);
+        public static void RegisterBeforeDispose(Func<ShellScope, Task> callback, bool last = false) => Current?.BeforeDispose(callback, last);
```

Comment:
Since the parameters are now not completely trivial, please add `<param>` docs for both.

Reply:
Done.